design-proposals/cluster-orch-k3s.md

```diff
@@ -0,0 +1,135 @@
+# Design Proposal: Add suppport for K3s
+
+Author(s): Hyunsun Moon, Andy Bavier, Denisio Togashi
+
+Last updated: 4/28/25
+
+## Abstract
+
+This proposal aims to enhance the Cluster Orchestration by introducing K3s as an additional option for edge cluster deployment. Currently, the framework supports only RKE2. By incorporating K3s, a production-grade, CNCF-certified lightweight and simplified Kubernetes distribution, the framework will allow users to select the most appropriate Kubernetes distribution for their specific use cases and the resource constraints of their environments. K3s will be set as the default choice, as it meets the needs of most deployment scenarios with its efficient resource usage, while RKE2 will remain available for users requiring a full Kubernetes experience for more complex applications. This change not only enhances flexibility but also lays the groundwork for potentially integrating additional Kubernetes distributions in the future, further expanding the framework's versatility and adaptability to diverse deployment scenarios.
+
+## Proposal
+
+### User Stories (Need Update)
+
+#### Story 1
+
+As an edge application deployment engineer, I want to transition from Docker to Kubernetes without incurring additional overhead for running Kubernetes itself, so that I can allocate more resources to application workloads and leverage advanced container orchestration capabilities that Kubernetes provides.
+
+#### Story 2
+
+As an edge operator, I want to deploy Kubernetes clusters in environments with limited computational resources, such as remote IoT installations or small-scale edge sites, and efficiently manage and operate Kubernetes-based applications without exceeding resource capacities.
```

Comment:
just fyi https://www.linkedin.com/posts/ncresswell_anyone-keen-to-get-their-hands-on-a-new-project-activity-7317870483843031040-B3EB?utm_source=share&utm_medium=member_desktop&rcm=ACoAAByuWAEB5UgX2HA4bfzyETziKmU-3yb3b_w